### DISCUSSION TITLE
Looped reload of devtools - how to get rid of?

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4120] [General]
![image](https://user-images.githubusercontent.com/42960488/170668748-3145fef4-3252-44c3-a585-8a9a41eb5d09.png)
Launching devtools I get looped reload and refresh of the window. Why does that happen in winforms browser version?

**Comment by aimcosoftware** [comment]
I'm doing the same without any issue. I do check `IsBrowserInitialized` first.
```
Private Sub Browser_IsBrowserInitializedChanged(Sender As Object, e As EventArgs) Handles Browser.IsBrowserInitializedChanged
	If DeferDevTools And Browser.IsBrowserInitialized Then
		ShowDevTools()
	End If
End Sub
```

**Reply by amaitland** [reply]
If you are using WinForms then there is no need for the IsBrowserInitialized check in the 
IsBrowserInitializedChanged event. 

The event is only raised once after the underlying CEF browser instance has been created.

**Comment by amaitland** [comment]
I'd say there is a mistake in your code somewhere. Would need more detail to say more.

Please no images of code, markdown formatted code please. 

https://idownvotedbecau.se/imageofcode

**Reply by lofti198** [reply]
Ok I see. What you mean by markdown formatted code?

**Reply by amaitland** [reply]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by amaitland** [reply]
I'm wondering if your Form is attempting to take focus at the same time as devtools launches and causing some sort of loop.

You can try opening `DevTools` on the `UI Thread`.

```c#
private void BrowserIsBrowserInitializedChanged(object sender, EventArgs e)
{
	var browser = (ChromiumWebBrowser)sender;

	browser.BeginInvoke((Action)delegate
	{
		browser.ShowDevTools();
	});
}
```
